src/plugins/discover/public/application/components/top_nav/get_top_nav_links.ts

```diff
@@ -122,7 +123,13 @@ export const getTopNavLinks = ({
         objectType: 'search',
         sharingData: {
           ...sharingData,
-          title: savedSearch.title,
+          // CSV reports can be generated without a saved search so we provide a fallback title
+          title:
+            savedSearch.title ||
+            i18n.translate('discover.localMenu.fallbackReportTitle', {
+              defaultMessage: 'Discover search [{date}]',
+              values: { date: moment().toISOString() },
```

Comment:
Without true it would return UTC time, but I think it would be more useful to provide an ISO string with offset to UTC
```suggestion
              values: { date: moment().toISOString(true) },
```
BTW great improvement! love it that you no longer have to save a search to export!